uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Reminder.php

```diff
@@ -131,14 +131,19 @@ private static function _extractPreviousAssignee(&$assignees, $previousAssigneeI
    * Gets the full list of the recipients of the reminder email
    * Makes sure there are no duplicates or no empty emails in the list
    *
-   * @param array $activityContacts
+   * @param array $contacts
    * @param array $previousAssignee
    * @return array
    */
-  private static function _reminderRecipients($activityContacts, $previousAssignee) {
-    return array_filter(array_unique(array_merge(
-                            $activityContacts['assignees']['emails'], $activityContacts['source']['emails'], [$previousAssignee['email']]
-    )));
+  private static function _reminderRecipients($contacts, $previousAssignee) {
+    $hasSources = !empty($contacts['source']['emails']);
```

Comment:
yup I was talking about something like this :

```php
$assignees = !empty($contacts['assignees']['emails']) ? $contacts['assignees']['emails'] : [];
```

but I am fine with your version and keeping  line length less than 80 character as you mentioned on hipchat is good point actually .. but I think adding some spaces will make this a bit better :

```php
    $hasSources = !empty($contacts['source']['emails']);
    $hasAssignees = !empty($contacts['assignees']['emails']);

    $assignees = $hasAssignees ? $contacts['assignees']['emails'] : [];
    $sources = $hasSources ? $contacts['source']['emails'] : [];
    $previous = $previousAssignee ? [$previousAssignee] : [];

    $recipients = array_merge($assignees, $sources, $previous);
```